homeassistant/components/sensor/__init__.py

```diff
@@ -361,25 +361,30 @@ def capability_attributes(self) -> dict[str, Any] | None:
     def _is_valid_suggested_unit(self, suggested_unit_of_measurement: str) -> bool:
         """Validate the suggested unit.
 
-        Validate that a unit converter exists for the sensor's device class and that the
-        unit converter supports both the native and the suggested units of measurement.
+        Validate that the native unit of measurement can be converted to the
+        suggested unit of measurement, either because they are the same or
+        because a unit converter supports both.
         """
-        # Make sure we can convert the units
-        if self.native_unit_of_measurement != suggested_unit_of_measurement and (
-            (unit_converter := UNIT_CONVERTERS.get(self.device_class)) is None
-            or self.__native_unit_of_measurement_compat
-            not in unit_converter.VALID_UNITS
-            or suggested_unit_of_measurement not in unit_converter.VALID_UNITS
+        # No need to check the unit converter if the units are the same
+        if self.native_unit_of_measurement == suggested_unit_of_measurement:
+            return True
+
+        # Make sure there is a unit converter and it supports both units
+        if (
+            (unit_converter := UNIT_CONVERTERS.get(self.device_class))
+            and self.__native_unit_of_measurement_compat in unit_converter.VALID_UNITS
+            and suggested_unit_of_measurement in unit_converter.VALID_UNITS
         ):
```

Comment:
Note: I ended up converting both to early exits, which allows for positive "and + in" rather than negative "or + not in"